docker-bake.hcl

```
@@ -76,8 +90,33 @@ variable "DEBIAN_RELEASE" {
   default = "bookworm-20240513"
 }
 
-variable "default_jdk" {
-  default = 17
+variable "WINDOWS_FLAVORS_TO_BUILD" {
+  default = "[\"nanoserver\",\"windowsservercore\"]"
```

Could you explain why are there backslash on these? It's not really clear as I would suggest to write as a [normal HCL list](https://developer.hashicorp.com/terraform/language/expressions/types#list) such as:

```suggestion
  default = ["nanoserver","windowsservercore"]
```

---

The backslashes are part of the string WINDOWS_FLAVORS_TO_BUILD, this isn't an HCL array.

I need them to define the content I want in a JSON format, to be decoded later in an array of flavors to build.

This is the only way I found in order to be able to define array values values via environment variables and to interact with them from the outside.

---

I've explained (hopefully more clearly) this with a bit more details here: https://github.com/docker/buildx/discussions/2398#discussioncomment-9778966

Reworked this PR with two targets "nanoserver" and "windowsservercore" instead of the "allwindows", and kept only one WINDOWS_VERSION_OVERRIDE variable to be able to set a unique Windows version when calling docker buildx bake to generate the docker compose file.

---

That look really clear, thanks!